RecoLuminosity/LumiAnalyzer/plugins/LumiAnalyzer.cc

```diff
@@ -0,0 +1,416 @@
+
+   // Get the Run, Lumi Section, and Event numbers, etc.
+   runNo = iEvent.id().run();
+   LSNo = iEvent.getLuminosityBlock().luminosityBlock();
+   eventNo = iEvent.id().event();
+   BXNo = iEvent.bunchCrossing();
+   timeStamp = iEvent.time().unixTime();
+   orbitN = iEvent.orbitNumber();
+
+   // add the vertex information
+
+   nVtx = 0;
+   nGoodVtx = 0;
+   nTrk = 0;
+   xV = -9999.; yV = -9999.; zV = -9999.;  chi2 = -9999.;  ndof = -9999;
+   
+   if(includeVertexInformation){
+      edm::Handle<reco::VertexCollection> recVtxs;
+      iEvent.getByLabel("offlinePrimaryVertices",recVtxs);
+   
+      if(recVtxs.isValid()){
+	 for(reco::VertexCollection::const_iterator v=recVtxs->begin(); v!=recVtxs->end(); ++v){
+	    int nnTrk = v->tracksSize();
+	    if(qPrint) { std::cout << "\n ****Event: " << eventNo << " Vertex nnTrk=" << nnTrk << std::endl; }
+	    int nd = (int)v->ndof();
+	    if(nd > 4 && v->isValid() && (v->isFake() == 0)){
+	       nVtx++;
+	       if(nnTrk > 0){
+		  nGoodVtx++;
+                  if(nnTrk > nTrk){
+                     nTrk = nnTrk;                   
+		     xV = 10000.*v->x(); yV = 10000.*v->y(); zV = 10.*v->z();
+		     chi2 = v->chi2();
+		     ndof = (int)v->ndof();
+		     if(qPrint){ 
+			std::cout << "   Good nVtx=" << nVtx << " nGoodVtx=" << nGoodVtx << " nTrk=" << nTrk << " xV=" <<  xV  << " yV=" << yV << " zV=" << zV
+				  << " chi2=" << chi2 << " ndof=" << ndof << std::endl;
+		     } 
+		  }
+	       }
+	    }      
+	 }
+      }
+   }
+
+   // get the generalTracks information
+   // for now just do something very simple and count the number of
+   // of tracks with pT > 0.25 GeV
+   using reco::TrackCollection;
+
+   if(includeTracks){
+      edm::Handle<TrackCollection> tracks;
+      iEvent.getByLabel("generalTracks",tracks);
+
+      nGeneralTracks = 0;
+      n1GeV = 0;
+      n2GeV = 0;
+      for(TrackCollection::const_iterator itTrack = tracks->begin(); itTrack != tracks->end(); ++itTrack) {
+	 double pT = itTrack->pt();
+	 if(pT > 0.25)nGeneralTracks++;
+	 double eta = itTrack->eta();
+	 if(eta > -0.8 && eta < 0.8){
+	    if(pT > 1.0)n1GeV++;
+	    if(pT > 2.0)n2GeV++;
+	 }
+      }
+   }
+
+
+  if (0 == fInit) {
+    fInit = 1; 
+    // -- Setup cabling map and its map to detIDs
+    iSetup.get<SiPixelFedCablingMapRcd>().get(fCablingMap);
+    for (int i = 0; i < 40; ++i) {
+      fPFC[i] = new SiPixelFrameConverter(fCablingMap.product(), i);
+    }
+    
+    
+    edm::ESHandle<TrackerGeometry> pDD;
+    iSetup.get<TrackerDigiGeometryRecord>().get(pDD);
+
+    for (TrackerGeometry::DetContainer::const_iterator it = pDD->dets().begin(); it != pDD->dets().end(); it++){
+      if(dynamic_cast<PixelGeomDetUnit*>((*it))!=0){
+	DetId detId = (*it)->geographicalId();
+	uint32_t newDetId = detId;
+	
+	for (int fedid = 0; fedid < 40; ++fedid) {
+	  if (fPFC[fedid]->hasDetUnit(newDetId)) {
+	    fFEDID.insert(make_pair(newDetId, fedid)); 
+	    break;
+	  }
+	}
+      }
+    }
```

Comment:
Hi Slava,

Thanks for the feedback.  Indeed, this code could use a little cleaning.  I
hadn't bothered to do much of that for this pull request because I intended
to follow up with a newer version which is also less cluttered.

I will implement the cleaning you've suggested in this request and
forthcoming pull requests will be cleaner.

I am about to go on vacation for a week.  If I find a moment to sort this
out I'll let you know.  Otherwise, I'll take care of it the week after next.

Cheers,
Chris



On Fri, Feb 6, 2015 at 9:24 PM, Slava Krutelyov <notifications@github.com>
wrote:

> In RecoLuminosity/LumiAnalyzer/plugins/LumiAnalyzer.cc
> <https://github.com/cms-sw/cmssw/pull/6962#discussion_r24269174>:
>
> > +    edm::ESHandle<TrackerGeometry> pDD;
> > +    iSetup.get<TrackerDigiGeometryRecord>().get(pDD);
> > +
> > +    for (TrackerGeometry::DetContainer::const_iterator it = pDD->dets().begin(); it != pDD->dets().end(); it++){
> > +      if(dynamic_cast<PixelGeomDetUnit*>((*it))!=0){
> > +	DetId detId = (*it)->geographicalId();
> > +	uint32_t newDetId = detId;
> > +	
> > +	for (int fedid = 0; fedid < 40; ++fedid) {
> > +	  if (fPFC[fedid]->hasDetUnit(newDetId)) {
> > +	    fFEDID.insert(make_pair(newDetId, fedid));
> > +	    break;
> > +	  }
> > +	}
> > +      }
> > +    }
>
> ~up to 20 lines appear to compute something that's not needed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/6962/files#r24269174>.
>